fendermint/app/src/cmd/run.rs

```diff
@@ -375,3 +387,60 @@ fn to_address(sk: &SecretKey, kind: &AccountKind) -> anyhow::Result<Address> {
         AccountKind::Ethereum => Ok(Address::from(EthAddress::new_secp256k1(&pk)?)),
     }
 }
+
+async fn dispatch_resolver_events(
+    mut rx: tokio::sync::broadcast::Receiver<ResolverEvent<AppVote>>,
+    parent_finality_votes: VoteTally,
+    topdown_enabled: bool,
+) {
+    loop {
+        match rx.recv().await {
+            Ok(event) => match event {
+                ResolverEvent::ReceivedPreemptive(_, _) => {}
+                ResolverEvent::ReceivedVote(vote) => {
+                    dispatch_vote(*vote, &parent_finality_votes, topdown_enabled).await;
+                }
+            },
+            Err(RecvError::Lagged(n)) => {
+                tracing::warn!("the resolver service skipped {n} gossip events")
+            }
+            Err(RecvError::Closed) => {
+                tracing::error!("the resolver service stopped receiving gossip");
+                return;
+            }
+        }
+    }
+}
+
+async fn dispatch_vote(
+    vote: VoteRecord<AppVote>,
+    parent_finality_votes: &VoteTally,
+    topdown_enabled: bool,
+) {
+    match vote.content {
+        AppVote::ParentFinality(f) => {
+            if !topdown_enabled {
+                tracing::debug!("ignoring vote; topdown disabled");
```

Comment:
is this maybe too verbose as its inside a loop?

Reply:
Yeah, if we saw this it would be because:
1. We are the root subnet, but we are using the IPLD resolver to listen to things from one of the subnets we are interested in
2. Yes, somehow we are receiving votes about the finality of of our non-existing parent. Can this be someone trying to flood the system with spam?

I certainly didn't want to give the ability to someone to generate warnings and keep everyone up, but I thought if these arrive we should know because maybe we misconfigured something and we are ignoring these votes by mistake (more likely). It doesn't give any indication of who we should ban, so it's more like a "hey, this is very unexpected!" flag, with `debug` level in case we are already trying to figure out what's going on.

All I can think of is some kind of global bool to only warn once, but then it could be drowned out by other stuff. Maybe this way, if our node misses a handful of votes every 30 seconds, we can have an indication of why things aren't working. 

If they turn out to be spammy we should remove them.